### ISSUE TITLE
benchmarking: Most chunked_loss `forward()` isn't measured properly

### ISSUE THREAD

**Issue by Tcc0403** [linkedin/Liger-Kernel#789]
### 🐛 Describe the bug

Most benchmarking scripts for chunked loss family take initizations overhead (nn.Module instantiation, torch.compile, etc) into account in `forward` benchmarking. We should move initializations out of measurements and reuse the same modules instead.

I had it fixed in #643 but didn't get to merge the PR. 

Refer to: https://github.com/linkedin/Liger-Kernel/pull/643/files#diff-e7e51a82ce1e3b371c7917ccca4134d80c40f7d11ec58f907b8707856b086ff2

### Versions

liger-kernel==https://github.com/linkedin/Liger-Kernel/commit/0e379208951492d4823f1f723ac6844bf25eeeb2
